src/app/modules/ix-forms/components/ix-slide-in/ix-slide-in-ref.ts

```diff
@@ -0,0 +1,30 @@
+import { ComponentRef } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { IxSlideIn2Component } from 'app/modules/ix-forms/components/ix-slide-in/ix-slide-in2.component';
+import { IxSlideIn2Service } from 'app/services/ix-slide-in2.service';
+
+export class IxSlideInRef<T, D = unknown> {
+  slideInClosed$ = new Subject<D>();
+  componentRef: ComponentRef<T>;
+  id: string;
+
+  get componentInstance(): T {
+    return this.componentRef.instance;
+  }
+
+  constructor(
+    private slideInService: IxSlideIn2Service,
+    private slideIn2Component: IxSlideIn2Component,
+  ) {}
+
+  close(response?: D): void {
+    this.slideInClosed$.next(response);
+    this.slideInClosed$.complete();
+    this.slideInService.deleteRef(this.id);
```

Comment:
What if instead of calling `deleteRef`, we have service subscribe with `take(1)` to `slideInClosed$` for every slide in ref that was opened?

Reply:
In IxSlideInRef, I created a Subject onClose$ with which service will work

Reply:
Why do you need a new subject, why not just use `afterClosed`?